app/views/applications/edit.slim

```diff
@@ -0,0 +1,75 @@
+- title "Edit Application"
+- content_for(:header)
+  ol.breadcrumb.shadow
+    li.breadcrumb-item = link_to('Splits I/O', root_path)
+    li.breadcrumb-item = link_to('Settings', settings_path)
+    li.breadcrumb-item = link_to('Developer Applications', settings_path(anchor: 'applications'))
+    li.breadcrumb-item = link_to(@application.name, application_path(@application))
+    li.breadcrumb-item = link_to('Edit', edit_application_path(@application))
+- if current_user.nil?
+  article You need to be signed in to view this page.
+- else
+  article data-turbolinks-temporary=true
+    .card
+      .card-header.text-white Edit #{@application.name}
+      .card-body
+        .form-group
+          p
+            .row
+              .col-lg-8
+                input.form-control type='text' value=@application.uid disabled=true
+              .col-lg-4
+                b Client ID
+                i< = "(not changeable)"
+          p
+            .row
+              .col-lg-8
+                .input-group
+                  - if (Time.now - 24.hours) > @application.secret_generated_at
+                    input.form-control type='text' value="<Hidden forever>" disabled=true
+                  - else
+                    input.form-control.border-warning type='text' value=@application.secret disabled=true
+                  .input-group-append
+                    = button_to(application_path(@application), class: 'btn btn-warning', method: :patch, \
+                      params: {reset_secret: '1'}, \
+                      data: { \
+                        confirm: "Really reset the client secret for #{@application.name}? This will invalidate \
+                        the current secret, preventing it from being used. Existing user authorizations will not \
+                        be affected." \
```

Comment:
Not problem with this PR, but there should probably be a way to revoke all tokens and auths if resetting the secret doesn't do that.  If you do leak your secret there should be a way to make sure someone else cannot be requesting token with your previously set up apps.

Reply:
Ah, if the functionality exists that's probably fine to be honest.  I just know that accidents happen and I've accidentally committed keys before, so having a way to wipe all access tokens is good.